frontend/src/components/ColorLegend.svelte

```diff
@@ -1,28 +1,114 @@
 <script lang="ts">
-    export let colors;
+    export let styles;
+    import G6, { Graph } from "@antv/g6";
+    import { onMount } from "svelte";
+    import { createEventDispatcher } from "svelte";
+    import { defaultNode } from "../layout";
+    const options = {
+        container: "mountNode",
+        width: 280,
+        height: 220,
+        workerEnabled: true,
+        nodeSize: 20,
+    };
+    const dispatch = createEventDispatcher();
+
+    let data = {};
+    let array = [];
+    styles.map((event) => {
+        let node = {
+            id: event[0],
+            size: 10,
+            type: event[1].Shape,
+            style: {
+                fill: event[1].Color,
+                stroke: event[1].Color,
+            },
+            label: event[1].Acronym,
+            labelCfg: {
+                style: {
+                    fill: event[1].Color,
+                },
+                position: "bottom",
+                offset: 10,
+            },
+        };
+        array.push(node);
+    });
+    let dNode = {
+        id: "defaultNode",
+        size: 10,
+        type: defaultNode.shape,
+        style: {
+            fill: defaultNode.color,
+            stroke: defaultNode.color,
+        },
+        label: defaultNode.type,
+        labelCfg: {
+            style: {
+                fill: defaultNode.color,
+            },
+            position: "bottom",
+            offset: 10,
+        },
+    };
+    array.push(dNode);
+    data = { nodes: array };
```

Comment:
Please make this a function instead, it is not easy to understand for as to what this is doing.